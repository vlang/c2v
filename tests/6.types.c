#include <stdio.h>
#include <stdbool.h>
#include <stdint.h>

// standard types should be lower-cased
typedef unsigned short v_u16;
typedef float v_f32;
typedef size_t v_usize;
typedef void *v_voidptr;
typedef bool v_bool;

// disabled for macOS/Linux compatibility
//#include <stddef.h>
//typedef ptrdiff_t v_isize;

// not sure how this is supposed to work
typedef intptr_t c_intptr_t;

int main()
{
    void *pointers[8];

    return 0;
}
