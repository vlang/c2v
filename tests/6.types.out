[translated]
module main

type V_u16 = u16
type V_f32 = f32
type V_usize = usize
type V_voidptr = voidptr
type V_bool = bool
type C_intptr_t = C.intptr_t

fn main() {
	pointers := [8]voidptr{}
	return
}
